src/net_processing.cpp

```diff
@@ -2274,9 +2276,11 @@ bool PeerManagerImpl::MaybeSendGetHeaders(CNode& pfrom, const CBlockLocator& loc
 
     const auto current_time = NodeClock::now();
 
+    LOCK(peer.m_last_getheaders_mutex);
```

Comment:
Is it not better to first lock the mutex and then acquire the current time, because locking the mutex can take "long" time and then we would operate on an outdated `current_time`?